client/core/core.go

```diff
@@ -2035,6 +2035,101 @@ func (c *Core) RescanWallet(assetID uint32, force bool) error {
 	return nil
 }
 
+// RecoverWallet will retrieve some recovery information from the wallet,
+// which may not be possible if the wallet is too corrupted, disonnect and
+// destroy the old wallet, create a new one, and if the recovery information
+// was retrieved from the old wallet, send this information to the new one.
+// If force is false, this will check for active orders involving this
+// asset before initiating a rescan. WARNING: It is ill-advised to initiate
+// a wallet recovery with active orders unless the wallet db is definitely
+// corrupted and even a rescan will not save it.
+//
+// DO NOT MAKE CONCURRENT CALLS TO THIS FUNCTION WITH THE SAME ASSET.
+func (c *Core) RecoverWallet(assetID uint32, appPW []byte, force bool) error {
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return newError(authErr, "RecoverWallet password error: %v", err)
+	}
+
+	if !force {
+		for _, dc := range c.dexConnections() {
+			if dc.hasActiveAssetOrders(assetID) {
+				return newError(activeOrdersErr, "active orders for %v", unbip(assetID))
+			}
+		}
+	}
+
+	oldWallet, found := c.wallet(assetID)
+	if !found {
+		return fmt.Errorf("RecoverWallet: wallet not found for %d -> %s: %w",
+			assetID, unbip(assetID), err)
+	}
+
+	dbWallet, err := c.db.Wallet(oldWallet.dbID)
+	if err != nil {
+		return fmt.Errorf("error retreiving DB wallet: %w", err)
+	}
+
+	seed, pw, err := c.assetSeedAndPass(assetID, crypter)
+	if err != nil {
+		return err
+	}
+
+	gotRecoveryCfg := true
+	recoveryCfg, err := oldWallet.GetRecoveryCfg()
+	if err != nil {
+		gotRecoveryCfg = false
+		c.log.Errorf("RecoverWallet: unable to get recovery config: %v", err)
+	}
+
+	oldWallet.Disconnect()
+
+	if err = oldWallet.Destroy(); err != nil {
+		return err
+	}
+
+	if err = asset.CreateWallet(assetID, &asset.CreateWalletParams{
+		Type:     dbWallet.Type,
+		Seed:     seed,
+		Pass:     pw,
+		Settings: dbWallet.Settings,
+		DataDir:  c.assetDataDirectory(assetID),
+		Net:      c.net,
+		Logger:   c.log.SubLogger("CREATE"),
+	}); err != nil {
+		return fmt.Errorf("Error creating wallet: %w", err)
+	}
+
+	newWallet, err := c.loadWallet(dbWallet)
+	if err != nil {
+		return newError(walletErr, "error loading wallet for %d -> %s: %w",
+			assetID, unbip(assetID), err)
+	}
+
+	_, err = c.connectWallet(newWallet)
+	if err != nil {
+		return err
+	}
+
+	if gotRecoveryCfg {
+		err = newWallet.ApplyRecoveryCfg(recoveryCfg)
+		if err != nil {
+			c.log.Errorf("RecoverWallet: failed to apply recovery config: %v", err)
+		}
+	}
+
+	c.updateAssetWalletRefs(assetID, newWallet)
+
+	err = newWallet.Unlock(crypter)
+	if err != nil {
+		c.log.Errorf("Error unlocking wallet after recovery: %v", err)
+	}
```

Comment:
Maybe just return error on these two errors? If we are unable to unlock, that's definitely a problem.

Reply:
For `Unlock` I agree, but shouldn't `Unlock` still be called if `ApplyRecoveryCfg` fails?

Reply:
Well, ok. I guess maybe the old wallet gave us some bogus recovery config, then may error.